capstone/src/main/java/com/google/sps/servlets/SmallOwnedRestaurantsDataServlet.java

```diff
@@ -53,41 +66,61 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 
       for (Element restaurantName : restaurantNameElements) {
         String name = restaurantName.text();
-        detailedPlaces.add(details.request(name));
-
         restaurantNames.add(name);
       }
     }
+    return restaurantNames;
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String keywords = (String) request.getParameter("keyword");
+    Query query = new Query(DATABASE_NAME).addSort("rating", SortDirection.DESCENDING);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    List<Entity> allRestaurants = datastore.prepare(query).asList(FetchOptions.Builder.withDefaults());
```

Comment:
Same here. Pulling all the restaurants from the datastore into memory concerns me.